### ISSUE TITLE
[Azure AD] Rancher does not handle an error when getting groups - leading to panic

### ISSUE THREAD

**Issue by maxsokolovsky** [rancher/rancher#39908]
**Rancher Server Setup**
- Rancher version: v2.7-head and v2.6-head
- Installation option (Docker install/Helm Chart):
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc):
- Proxy/Cert Details:

**Information about the Cluster**
- Kubernetes version:
- Cluster Type (Local/Downstream):
   - If downstream, what type of cluster? (Custom/Imported or specify provider for Hosted/Infrastructure Provider):

**User Information**
- What is the role of the user logged in? (Admin/Cluster Owner/Cluster Member/Project Owner/Project Member/Custom)
  - If custom, define the set of permissions:

**Describe the bug**
If admins do not grant sufficient permissions to their apps in Azure, such that Rancher does not have enough permissions to get groups, Rancher panics (although does not crash the server) instead of reporting an error. To the user, this all looks like an expected 500, but for the wrong reason. The panic is in the logs because Rancher does not handle the permissions error.

**To Reproduce**
Grant to your Azure app only `Group.Read.All` permission. Set up Azure AD in Rancher and login. Try searching for principals when adding project members. Observe search fails with a 500, which is fine. But there is a panic in the logs.

**Result**

**Expected Result**

**Screenshots**

**Additional context**

**Comment by maxsokolovsky** [comment]
# QA Testing for v2.6 and v2.7.

1. Go to the Azure portal and set your app's permissions to:
`User.Read.All`
2. Remove any other permissions.
3. Enable Azure AD as the auth provider in Rancher.
4. Notice that you can log in and out fine.
5. Add a new project and search for users to add to it.
6. The search will fail - there will be no results, and a 500 error in the console will appear.
7. Check Rancher logs and ensure there are no panics.

**Comment by dasarinaidu** [comment]
This issue is fixed on  `v2.7-c24fb8b0869a0b445f55b3307c6ed4582e147747-head`

I did not see any panic logs when I disable Group.Read.All

**Steps:**

1. Pre-condition: At AZURE portal make sure the settings

```
Group.Read.All -- Disabled
User.Read.All -- Enabled
```

2. Rancher -> Auth Provider (AZURE)
3. Connect to AZURE
4. Local cluster -> Project/Cluster members -> Try to search for a Group
5. Check for the Network and rancher Logs

**Actual:**
a. No Users/Groups were searchable
b. Network call throws 500 Error https://aaaaaaaaaa/v3/principals?action=search
c. Rancher Logs were NOT panic and giving appropriate error.

```
2023/03/02 16:39:40 [ERROR] Unknown error: server error while authenticating: UsersClient.BaseClient.Get(): unexpected status 403 with OData error: Authorization_RequestDenied: Insufficient privileges to complete the operation.
2023/03/02 16:40:15 [INFO] Failed to decode the 'metadata' field of the AuthConfig. Attempting to decode 'annotations' at the top level.
2023/03/02 16:40:42 [ERROR] Unknown error: GroupsClient.BaseClient.Get(): unexpected status 403 with OData error
```
